### PR TITLE
Cycle sort in c#

### LINKED ISSUE #1672: Add Cycle sort in C#
I would like to work on cycle sort in C#, as part of GSSoC 20 . So Please assign this to me

### PR DESCRIPTION
**Pull request by Harikrishnan6336**
Fixes #1672 

#### Checklist:
- [x] 4 space indentation.
- [x] Coding conventions are followed.
- [x] Input is taken dynamically.
- [x] Sample Input / Output is added at the end of file.
- [x] Logic Documentation (Comments).
- [x] File names are correct.

Make sure these boxes are checked before your pull request (PR) is ready to be reviewed and merged. Thanks!

#### Changes proposed in this pull request:
- Added Cycle-sort-in C#
- 

#### Languages Used:
- C#
- 

#### Files Added:
- Cycle_sort.cs
- 

> We're happy to help you get this ready -- don't be afraid to ask for help.

Thanks!

### CODE REVIEW

File: Cycle_Sort/Cycle_Sort.cs [hunk 1]
```diff
@@ -0,0 +1,88 @@
...
+                    int temp = item; 
+                    item = arr[pos]; 
+                    arr[pos] = temp; 
+                    writes++; 
+                } 
+            } 
+        } 
+    } 
+  
+public static void Main()
```
Comment (asha15) [inline Cycle_Sort/Cycle_Sort.cs]: can you keep the indentation

File: Cycle_Sort/Cycle_Sort.cs [hunk 2]
```diff
@@ -0,0 +1,88 @@
...
+                    writes++; 
+                } 
+            } 
+        } 
+    } 
+  
+public static void Main() 
+    { 
+        int size;
+        size= Convert.ToInt32(Console.ReadLine());
```
Comment (asha15) [inline Cycle_Sort/Cycle_Sort.cs]: keep a space before =

File: Cycle_Sort/Cycle_Sort.cs [hunk 3]
```diff
@@ -0,0 +1,88 @@
...
+            } 
+        } 
+    } 
+  
+public static void Main() 
+    { 
+        int size;
+        size= Convert.ToInt32(Console.ReadLine());  
+        int[] array = new int[size];
+        for (int i=0; i<size; i++)
```
Comment (asha15) [inline Cycle_Sort/Cycle_Sort.cs]: keep space before and after operators

File: Cycle_Sort/Cycle_Sort.cs [hunk 4]
```diff
@@ -0,0 +1,88 @@
...
+    { 
+        int size;
+        size= Convert.ToInt32(Console.ReadLine());  
+        int[] array = new int[size];
+        for (int i=0; i<size; i++) 
+    	      array[i] = Convert.ToInt32(Console.ReadLine());
+        cycleSort(array, size); 
+        Console.Write("After sorting  : ");
+        for (int i = 0; i < size; i++) 
+            Console.Write(array[i] + " ");
```
Comment (asha15) [inline Cycle_Sort/Cycle_Sort.cs]: please correct the indentation

File: Cycle_Sort/Cycle_Sort.cs [hunk 5]
```diff
@@ -0,0 +1,88 @@
...
+        for (int i=0; i<size; i++) 
+    	      array[i] = Convert.ToInt32(Console.ReadLine());
+        cycleSort(array, size); 
+        Console.Write("After sorting  : ");
+        for (int i = 0; i < size; i++) 
+            Console.Write(array[i] + " "); 
+    } 
+} 
+//SAMPLE INPUT AND OUTPUT
+/* 7
```
Comment (asha15) [inline Cycle_Sort/Cycle_Sort.cs]: it's better if you declaring this as number of elements or something

File: Cycle_Sort/Cycle_Sort.cs [hunk 6]
```diff
@@ -0,0 +1,88 @@
...
+//SAMPLE INPUT AND OUTPUT
+/* 7 
+ 4
+ 9 
+ 3
+ 9
+ 4
+ 7
+ 5
+After sorting : 3 4 4 5 7 9 9
```
Comment (asha15) [inline Cycle_Sort/Cycle_Sort.cs]: please clearly declare this as input and output

File: Cycle_Sort/Cycle_Sort.cs [hunk 7]
```diff
@@ -0,0 +1,88 @@
...
+ 7
+ 5
+After sorting : 3 4 4 5 7 9 9 
+ 5
+ 5 
+ 1 
+ 7 
+ 3 
+ 2
+ After sorting : 1 2 3 5 7
```
Comment (asha15) [inline Cycle_Sort/Cycle_Sort.cs]: clearly state this as second exmaple

File: Cycle_Sort/Cycle_Sort.cs (hunk 4 above)
Comment (Harikrishnan6336) [reply Cycle_Sort/Cycle_Sort.cs]: What is wrong here ??

File: Cycle_Sort/Cycle_Sort.cs [hunk 8]
```diff
@@ -0,0 +1,94 @@
...
+        int[] array = new int[size];
+        Console.Write("\nEnter the numbers : ");
+        for (int i = 0; i < size; i++) 
+    	      array[i] = Convert.ToInt32(Console.ReadLine());
+        cycleSort(array, size); 
+        Console.Write("\nAfter sorting  : ");
+        for (int i = 0; i < size; i++) 
+            Console.Write(array[i] + " "); 
+    } 
+}
```
Comment (asha15) [inline Cycle_Sort/Cycle_Sort.cs]: please a line space after this. Then will go for merge

File: Cycle_Sort/Cycle_Sort.cs [hunk 9]
```diff
@@ -0,0 +1,95 @@
...
+        cycleSort(array, size); 
+        Console.Write("\nAfter sorting  : ");
+        for (int i = 0; i < size; i++) 
+            Console.Write(array[i] + " "); 
+    } 
+} 
+
+//SAMPLE INPUT AND OUTPUT
+/*SAMPLE 1
+//Enter the number of numbers : 7
```
Comment (asha15) [inline Cycle_Sort/Cycle_Sort.cs]: I feel the naming is not suitable, can you change it ? the will go for merge

Comment (Harikrishnan6336) [reply Cycle_Sort/Cycle_Sort.cs]: @asha15 Done..

**Comment by vaishnavi-janardhan** [comment]
@asha15 Please modify the label `GSSoC 20` to `gssoc20`

**Pull request merged**